folium/map.py

```diff
@@ -322,6 +321,24 @@ def __init__(
             **kwargs,
         )
 
+    class SetIcon(MacroElement):
+        """Set the icon of a marker after both are created."""
+        _template = Template("""
+            {% macro script(this, kwargs) %}
+                {{ this.marker.get_name() }}.setIcon({{ this.icon.get_name() }});
+            {% endmacro %}
+        """)
+
+        def __init__(self, marker: 'Marker', icon: 'Icon'):
+            super().__init__()
+            self._name = "SetIcon"
+            self.marker = marker
+            self.icon = icon
+
+    def register_marker(self, marker: "Marker") -> None:
```

Comment:
To me it would feel more natural to have a `setIcon` method on `Marker` (which does the same thing).